components/TextFields/tests/unit/ContainedInputView/MDCBaseTextFieldTests.m

```diff
@@ -32,71 +34,457 @@ - (UIView *)createSideView {
   textField.trailingView = sideView;
+
+  // Then
   XCTAssertTrue(textField.rightView == textField.trailingView,
                 @"The trailing view should be the right view.");
 }
 
 - (void)testTrailingViewRTL {
+  // Given
   MDCBaseTextField *textField = [[MDCBaseTextField alloc] initWithFrame:CGRectMake(0, 0, 100, 60)];
   textField.layoutDirection = UIUserInterfaceLayoutDirectionRightToLeft;
+
+  // When
   UIView *sideView = [self createSideView];
   textField.trailingView = sideView;
+
+  // Then
   XCTAssertTrue(textField.leftView == textField.trailingView,
                 @"The trailing view should be the left view.");
 }
 
 - (void)testLeadingViewModeLTR {
+  // Given
   MDCBaseTextField *textField = [[MDCBaseTextField alloc] initWithFrame:CGRectMake(0, 0, 100, 60)];
+
+  // When
   textField.layoutDirection = UIUserInterfaceLayoutDirectionLeftToRight;
   textField.leadingViewMode = UITextFieldViewModeAlways;
+
+  // Then
   XCTAssertTrue(textField.leftViewMode == textField.leadingViewMode,
                 @"The leading view mode should be equal to the left view mode.");
 }
 
 - (void)testLeadingViewModeRTL {
+  // Given
   MDCBaseTextField *textField = [[MDCBaseTextField alloc] initWithFrame:CGRectMake(0, 0, 100, 60)];
+
+  // When
   textField.layoutDirection = UIUserInterfaceLayoutDirectionRightToLeft;
   textField.leadingViewMode = UITextFieldViewModeAlways;
+
+  // Then
   XCTAssertTrue(textField.rightViewMode == textField.leadingViewMode,
                 @"The leading view mode should be equal to the right view mode.");
 }
 
 - (void)testTrailingViewModeLTR {
+  // Given
   MDCBaseTextField *textField = [[MDCBaseTextField alloc] initWithFrame:CGRectMake(0, 0, 100, 60)];
+
+  // When
   textField.layoutDirection = UIUserInterfaceLayoutDirectionLeftToRight;
   textField.trailingViewMode = UITextFieldViewModeAlways;
+
+  // Then
   XCTAssertTrue(textField.rightView == textField.trailingView,
                 @"The trailing view mode should be equal to the right view mode.");
 }
 
 - (void)testTrailingViewModeRTL {
+  // Given
   MDCBaseTextField *textField = [[MDCBaseTextField alloc] initWithFrame:CGRectMake(0, 0, 100, 60)];
+
+  // When
   textField.layoutDirection = UIUserInterfaceLayoutDirectionRightToLeft;
   textField.trailingViewMode = UITextFieldViewModeAlways;
+
+  // Then
   XCTAssertTrue(textField.leftViewMode == textField.trailingViewMode,
                 @"The trailing view mode should be equal to left view mode.");
 }
 
+- (void)testAdjustTextAreaFrameWithParentClassTextAreaFrame {
+  // Given
+  CGRect textAreaFrame = CGRectMake(30, 50, 120, 20);
+  CGRect parentClassTextAreaFrame = CGRectMake(20, 30, 120, 50);
+
+  // When
+  CGRect finalTextAreaFrame =
+      [[[MDCBaseTextField alloc] init] adjustTextAreaFrame:textAreaFrame
+                              withParentClassTextAreaFrame:parentClassTextAreaFrame];
+
+  // Then
+  CGRect correctFinalTextAreaFrame = CGRectMake(30, 35, 120, 50);
```

Comment:
How were these values chosen? Please either document or, preferably, compute them here so that if the test fails it's clear where these numbers came from.

Reply:
Done